### ISSUE TITLE
Volunteer  Foreign Key Contstraint Error

### ISSUE THREAD

**Issue by chazkyle** [HTBox/crisischeckin#402]
REPO: 
https://crisischeckin-d.azurewebsites.net/Home

Choose Disaster that doesn't have an activity. Enter valid date range and location. Click Volunteer

ERROR:
Server Error in '/' Application.
The INSERT statement conflicted with the FOREIGN KEY constraint "FK_dbo.Commitment_dbo.Cluster_ClusterId". The conflict occurred in database "crisischeckin-dev", table "dbo.Cluster", column 'Id'.
The statement has been terminated.
Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.Data.SqlClient.SqlException: The INSERT statement conflicted with the FOREIGN KEY constraint "FK_dbo.Commitment_dbo.Cluster_ClusterId". The conflict occurred in database "crisischeckin-dev", table "dbo.Cluster", column 'Id'.
The statement has been terminated.

Source Error:

An unhandled exception was generated during the execution of the current web request. Information regarding the origin and location of the exception can be identified using the exception stack trace below.

**Comment by mjmilan** [comment]
The scenario described is something we ought to stop the user from being able to do. I don't think it makes sense to have them volunteer but not specify which cluster / activity their expertise lies within...

That said, I am a newcomer to the project - you'd do well to take steer from @BillWagner ...

**Comment by mjmilan** [comment]
@chazkyle 

Is this still a valid issue Charles?

The reason I ask is that I am new to the project, so I've been looking for this "Actvity" that Disasters are supposed to have, but I can't find any trace of them. I am proceeding under the assumption that at some point before I started looking at the project they were renamed to Clusters, though I'm not sure about that because the exception text you have supplied clearly refers to Clusters as well.

If we can assume Clusters and Activities to be one and the same, then I am finding it impossible through normal use of the application to create a Disaster with no Cluster selected - the validation is checking that we have at least one Cluster selected.

In short, I think this can probably be closed, but this should be a decision taken by either @chazkyle , @tonysurma , @BillWagner or @RichardCampbell .
